components/ciphertool/src/main/java/org/wso2/ciphertool/utils/KeyStoreUtil.java

```diff
@@ -31,6 +32,42 @@
 
 public class KeyStoreUtil {
 
+    /**
+     * Retrieves the password for the keystore.
+     *
+     * @return The keystore password.
+     */
+    public static String getKeystorePassword() {
+
+        String password;
+        String keyStoreName = ((Utils.isPrimaryKeyStore()) ? "Primary" : "Internal");
+        if (StringUtils.isNotEmpty(System.getProperty(Constants.KEYSTORE_PASSWORD))) {
+            password = System.getProperty(Constants.KEYSTORE_PASSWORD);
+        } else {
+            password = Utils.getValueFromConsole("Please Enter " + keyStoreName + " KeyStore Password of Carbon Server : ", true);
+            System.setProperty(Constants.KEYSTORE_PASSWORD, password);
+        }
+        if (password == null) {

```

Comment:
If the program stops at "password == null," will the journey to get back here be lengthy? if so lets optimize the journey by allowing reentering

Reply:
No, the password it entered at the beginning. Also this was the original behaviour